pandas/core/frame.py

```diff
@@ -7890,34 +7890,30 @@ def _count_level(self, level, axis=0, numeric_only=False):
                 f"Can only count levels on hierarchical {self._get_axis_name(axis)}."
             )
 
-        if frame._is_mixed_type:
-            # Since we have mixed types, calling notna(frame.values) might
-            # upcast everything to object
-            mask = notna(frame).values
-        else:
-            # But use the speedup when we have homogeneous dtypes
-            mask = notna(frame.values)
+        # Mask NaNs: Mask rows or columns where the index level is NaN, and all
+        # values in the DataFrame that are NaN
+        values_mask = notna(frame.values)
```

Comment:
is the existing version with `notna(frame).values` wrong?  avoiding .values with is_mixed_type is worthwhile

Reply:
@jbrockmendel I understood @jreback 's earlier [comment](https://github.com/pandas-dev/pandas/pull/32842#discussion_r396028090) such that `notna(frame.values)` and `notna(frame).values` are the same.

Reply:
those two are in fact the same, but performance should be better for `notna(frame).values`

Reply:
Running some tests shows that `notna(df).values` is indeed much faster on larger mixed type `DataFrames`. I will restore the original code. 
I guess this optimisation would be better in `notna`.
```python
In [1]: import pandas as pd                                                                                                                                                                   
In [2]: from pandas.core.dtypes.missing import notna                                                                                                                               
In [3]: df = pd.DataFrame({'a': range(10), 'b': ['foo'] * 10})                                                                                                                                     
In [4]: %timeit notna(df.values)                                                                                                                                                              
113 µs ± 9.55 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
In [5]: %timeit notna(df).values                                                                                                                                                              
546 µs ± 5.22 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
In [6]: df = pd.DataFrame({'a': range(1000000), 'b': ['foo'] * 1000000})                                                                                                                      
In [7]: %timeit notna(df.values)                                                                                                                                                              
163 ms ± 2.35 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
In [8]: %timeit notna(df).values                                                                                                                                                              
40.7 ms ± 291 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
In [9]: %timeit notna(df.values)                                                                                                                                                              
158 ms ± 1.7 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
In [10]: %timeit notna(df).values                                                                                                                                                             
39.7 ms ± 1.39 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```